src/box/sql/resolve.c

```diff
@@ -377,13 +410,17 @@ lookupName(Parse * pParse,	/* The parsing context */
 							 ER_SQL_PARSER_GENERIC,
 							 tt_sprintf(err, zAs));
 						pParse->is_aborted = true;
+						sql_xfree(old_tab);
+						sql_xfree(old_col);
 						return WRC_Abort;
 					}
 					if (sqlExprVectorSize(pOrig) != 1) {
 						diag_set(ClientError,
 							 ER_SQL_PARSER_GENERIC,
 							 "row value misused");
 						pParse->is_aborted = true;
+						sql_xfree(old_tab);
+						sql_xfree(old_col);
 						return WRC_Abort;
```

Comment:
I suggest to move that into the end of the function and jump there using `goto`. Duplicating this code for multiple returns doesn't look good.

Reply:
Fixed.